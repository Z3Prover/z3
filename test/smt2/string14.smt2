; Copyright (c) 2017 Microsoft Corporation
; issue #1015

(set-option :smt.string_solver seq) 
(declare-fun s() String)
(assert (= "\x80" s))
(check-sat)
(get-value (s))
(assert (= (str.len s) 1))
(check-sat)
(declare-fun c() String)
(assert (= (str.at s 0) c))
(check-sat)
(reset)

(set-option :smt.string_solver z3str3) 
(declare-fun s() String)
(assert (= "\x80" s))
(check-sat)
(get-value (s))
(assert (= (str.len s) 1))
(check-sat)
(declare-fun c() String)
(assert (= (str.at s 0) c))
(check-sat)

