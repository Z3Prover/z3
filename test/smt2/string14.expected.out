sat
((s "\x80"))
sat
sat
sat
((s "\x80"))
sat
sat
